### PR TITLE
chore: speed up CI a bit (hopefully)

### PR DESCRIPTION
**Pull request by gtm-nayan**
### Please don't delete this checklist! Before submitting the PR, please make sure you do the following:
- [ ] It's really useful if your PR references an issue where it is discussed ahead of time. In many cases, features are absent for a reason. For large changes, please create an RFC: https://github.com/sveltejs/rfcs
- [ ] This message body should clearly illustrate what problems it solves.
- [ ] Ideally, include a test that fails without this PR but passes with it.

### Tests
- [ ] Run the tests with `pnpm test` and lint the project with `pnpm lint` and `pnpm check`

### Changesets
- [ ] If your PR makes a change that should be noted in one or more packages' changelogs, generate a changeset by running `pnpm changeset` and following the prompts. Changesets that add features should be `minor` and those that fix bugs should be `patch`. Please prefix changeset messages with `feat:`, `fix:`, or `chore:`.

### CODE REVIEW

**Comment by gtm-nayan** [comment]
Aight, CI timings are all over the board so can't compare, but locally disabling tracing and disabling the TS loader for playwright got me speed ups of about 15ish percent, so this should theoretically be faster

And apparently disabling tracing causes flakes in a few tests for no apparent reason at all, so that's what the test changes are about.

File: packages/package/.gitignore [hunk 1]
```diff
@@ -6,6 +6,8 @@
 !/test/node_modules
 /test/apps/basics/test/errors.json
 .custom-out-dir
+test/watch/src/lib/*
+!test/watch/src/lib/index.js
```
Comment (benmccann) [inline packages/package/.gitignore:10]: why are these lines needed? I don't see "watch" anywhere else in this diff

also, I notice the other lines have a leading `/` whereas this one doesn't

File: packages/kit/test/utils.js [hunk 2]
```diff
@@ -261,7 +271,7 @@ export const config = {
 	use: {
 		...test_browser_device,
 		screenshot: 'only-on-failure',
-		trace: 'retain-on-failure'
+		trace: 'on-first-retry'
```
Comment (benmccann) [inline packages/kit/test/utils.js:274]: `PW_DISABLE_TS_ESM` seems like a good idea if it speeds things up, but I'm a bit less excited about this one because it will make flaky tests a lot harder to debug as we'll only get traces if they fail twice in a row. any idea how much of the speed difference if from `PW_DISABLE_TS_ESM` vs this?

File: packages/package/.gitignore (hunk 1 above)
Comment (gtm-nayan) [reply packages/package/.gitignore:10]: The package test creates some temporary files that aren't cleaned up if the test gets interrupted.

I'll change the slash.

File: packages/kit/test/utils.js (hunk 2 above)
Comment (gtm-nayan) [reply packages/kit/test/utils.js:274]: When just running tests from single packages, disabling tracing is about 4x more significant than the loader. I have to check how it adds up as a whole.

I thought people could reenable tracing through the CLI flag when debugging flaky tests but thinking again, that's near impossible without going into the test directories and flakiness might not repro locally. Maybe it's not worth it, yeah.

Comment (gtm-nayan) [reply packages/kit/test/utils.js:274]: Hmm, I just noticed that the condition for uploading traces is action failure, it doesn't upload them for flaky tests that pass on a retry, so it doesn't really change anything.

**Comment by Rich-Harris** [comment]
I don't really understand most of these changes, but is this ready to go?

File: packages/kit/test/utils.js (hunk 2 above)
Comment (benmccann) [reply packages/kit/test/utils.js:274]: > I just noticed that the condition for uploading traces is action failure

It might be a good idea to change that and upload them whenever they exist

> it doesn't upload them for flaky tests that pass on a retry, so it doesn't really change anything

I think it'd still change it a fair amount. E.g. if you have one test that fails three times in a row and ten tests that fail once and then pass, you'd only get the trace for the one test

**Comment by benmccann** [comment]
> A small problem I see now that I think about it is that now an error in cookies might cascade to the tests affected by 3, but if that's okay then yes it's ready to go.

I think each test gets its own browser, so cookies probably shouldn't be shared across tests?

Love the flakiness cleanup! I'm a bit worried about having less tracing still as I think it'll make it harder to identify and debug flakiness

**Review by Rich-Harris — APPROVED** [review]

**Comment by gtm-nayan** [comment]
I got about 15% improvements on local but I'm going through the past runs for this PR and the timing swings seem more luck based than anything else. Too much noise to make any claims of improvement. I'll re-enable tracing.

File: packages/kit/test/apps/basics/src/global.d.ts [hunk 3]
```diff
@@ -6,6 +6,7 @@ declare global {
 		mounted: number;
 		fulfil_navigation: (value: any) => void;
 		promise: Promise<any>;
+		update_state: number;
```
Comment (benmccann) [inline packages/kit/test/apps/basics/src/global.d.ts:9]: maybe `state_counter` would be a better name?

**Comment by benmccann** [comment]
The other thing that concerns me a bit is that there are a few places where `waitUntil: 'networkidle'` is being used and I wonder if we can avoid that or at least add comments as to why it's necessary

Screenshot from 2023-05-17 08-39-12

**Comment by gtm-nayan** [comment]
Ahh, some of those can be removed but some will have to stay, mostly those in tests with page.on request cause otherwise there'll be favicon/js requests that trip those up, but it might be possible to improve the filters there. I'll see tomorrow.

**Comment by dummdidumm** [comment]
@gtm-nayan care to take another look at this (merging master in) and confirm this all still works and give a thumbs up if this can go in?

File: packages/kit/test/apps/basics/test/client.test.js [hunk 4]
```diff
@@ -143,35 +143,42 @@ test.describe('Load', () => {
 	});
 
 	test('load does not call fetch if max-age allows it', async ({ page }) => {
-		page.addInitScript(`
+		// update_state is used to ensure we start waiting at the right time
```
Comment (benmccann) [inline packages/kit/test/apps/basics/test/client.test.js:146]: I'm not sure what this comment is referring to in terms of `update_state`

File: packages/kit/test/apps/basics/test/cross-platform/client.test.js [hunk 5]
```diff
@@ -75,7 +75,10 @@ test.describe('a11y', () => {
 			})
 		).toBe(1);
 
-		await clicknav('[href="/selection/b"]');
+		await clicknav('[href="/selection/b"]', {
+			waitUntil: 'networkidle'
```
Comment (benmccann) [inline packages/kit/test/apps/basics/test/cross-platform/client.test.js:79]: why do we need this?

**Comment by gtm-nayan** [comment]
forgot the context around some of the changes, I'll take a look tomorrow, the test failures are from master CI being broken i think

**Comment by benmccann** [comment]
@gtm-nayan I think we were getting close on this PR if you want to try giving it another look

**Comment by dummdidumm** [comment]
We should target version-2 for this (or wait until version-2 is on main)

**Comment by Rich-Harris** [comment]
is this ready? what's the tl;dr on what it's actually doing?

**Comment by benmccann** [comment]
As I recall, I had several points of feedback that weren't addressed yet. I don't know if @gtm-nayan still wants to continue with this or not? If not, perhaps we can split the flakiness fixes out into a new PR and close this one as I'm not sure we want to merge everything that was in this one though my memory of it could be outdated

File: packages/kit/test/apps/basics/test/cross-platform/client.test.js [hunk 6]
```diff
@@ -75,7 +75,8 @@ test.describe('a11y', () => {
 			})
 		).toBe(1);
 
-		await clicknav('[href="/selection/b"]');
+		await clicknav('0[href="/selection/b"]');
```
Comment (benmccann) [inline packages/kit/test/apps/basics/test/cross-platform/client.test.js:78]:
```suggestion
		await clicknav('[href="/selection/b"]');
```

**Comment by benmccann** [comment]
I removed some of the parts I was less sure of to unblock this and get the rest in. Feel free to resend those parts in a new PR with some more explanation
